src/auth/user.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@nestjs/common';
+import { AuthDto } from './dto/auth.dto';
+import { UserRepository } from './user.repisitory';
+import { UserEntity } from './user.entity';
+
+@Injectable()
+export class UserService {
+    constructor(
+         private readonly userRepository: UserRepository,
+    ) {}
+
+    async register(user: AuthDto): Promise<Omit<UserEntity, 'passwordHash'> > {
+        const registeredUser = await this.userRepository.createAndSaveUser(user);
+        const res = {
+            id: registeredUser.id,
+            name: registeredUser.name,
+            email: registeredUser.email,
+            created_at: registeredUser.created_at,
+            updated_at: registeredUser.updated_at,
+        }
+        return res;
+    }
```

Comment:
it is better to describe the response object with a separate class (with swagger decorators) and move its formation into a separate function